qm2/scf.py

```diff
@@ -76,38 +29,77 @@ def fock_build(D, damp=False):
     F_old = F_new
     return F
 
-
-for iteration in range(25):
-    F = fock_build(D, iteration>5) 
-
-    #Build the AO Gradient
-    grad = F @ D @ S - S @ D @ F      
-
-    grad_rms = np.mean(grad ** 2) ** 0.5
-
-    E_electric = np.sum((F+H) * D)
-    E_total = E_electric + mol.nuclear_repulsion_energy()
-
-    E_diff = E_total - E_old
-    E_old = E_total
-
-    print("Iter=%3d  E = % 16.12f  E_diff = % 8.4e  D_diff = % 8.4e" %
-            (iteration, E_total, E_diff, grad_rms))
-
-    #Break if e_conv and d_conv are met
-    if (abs(E_diff) < e_conv) and (grad_rms < d_conv):
-        break
-
-    eps, C = diag(F, A)
-    Cocc = C[:, :nel]
-    D = Cocc @ Cocc.T
-
-print("SCF has finished!\m")
-
-psi4.set_options({"scf_type": "pk"})
-psi4_energy = psi4.energy("SCF/aug-cc-pvdz", molecule=mol)
-print("Energy matches Psi4 %s" % np.allclose(psi4_energy, E_total))
-
-
-
-
+def run_scf(mol_string): 
+    e_conv = 1.e-6
+    d_conv = 1.e-6
+    nel = 5
+    damp_value = 0.20
+    damp_start = 5
+    
+    # Build a basis
+    bas = psi4.core.BasisSet.build(mol,target="aug-cc-pvdz")
```

Comment:
We need to give `mol` a value before this line. So, we can do that by using the `build_geom` function we have defined above.